test/unit-test/test_keystore.c

```diff
@@ -209,6 +209,17 @@ static void _test_keystore_get_xpub(void** state)
     wally_free_string(xpub_string);
 }
 
+static void _test_keystore_get_root_fingerprint(void** state)
+{
+    mock_state(_mock_seed, _mock_bip39_seed);
+    char fingerprint[4];
+    assert_true(get_root_fingerprint(fingerprint, sizeof(fingerprint)));
+    assert_int_equal(fingerprint[0], 0xffffffffffffffca);
+    assert_int_equal(fingerprint[1], 0x6d);
+    assert_int_equal(fingerprint[2], 0xffffffffffffff8b);
+    assert_int_equal(fingerprint[3], 0);
```

Comment:
did you check this against expected fingerprints from other libraries? this is probably the null temrinator from your snprintf, not the real fingerprint.